Automattic/wp-calypso
# EditGravatar: Add image editor step

This PR adds the ability to edit and crop an image to a square before uploading it to Gravatar. Here's what it looks like:

![image-editor-v2](https://cloud.githubusercontent.com/assets/11487924/20232263/45a390ac-a867-11e6-8906-81708764a695.gif)

## Steps to test

1. Run using `ENABLE_FEATURES=me/edit-gravatar make run`
2. Get a bearer token using this script:
    ```
    curl -# -X POST --data-urlencode "grant_type=password" --data-urlencode "client_id=appid" --data-urlencode "client_secret=appsecret" --data-urlencode "username=username" --data-urlencode "password=password" https://public-api.wordpress.com/oauth2/token
    ```
    You can use Calypso's app ID & secret, found [here](https://github.com/Automattic/wp-calypso/blob/a1c610c300e040a2c101bb5af6a66f7cd72b52d6/config/production.json#L108-L109).
    If you have 2FA enabled, you can add this extra parameter to the call: `--data-urlencode     "wpcom_otp=one-time-password"`, or you can create a new application password to use instead.
3. Save the bearer token to localStorage using `localStorage.setItem('bearerToken', 'your_bearer_token');`
4. Go to `/me` and upload a new Gravatar, see that you can edit it before uploading

## Still to do:

- [x] Add error handling for unsupported file extensions
- [x] Add error handling for response from image editor
